### PR TITLE
Do not print statements by insurance company

### PR DESCRIPTION
**Pull request by teryhill**
Do not print statements by insurance company #845

### CODE REVIEW

File: library/globals.inc.php [hunk 1]
```diff
@@ -1144,6 +1144,19 @@
       xl('Total Minimum Dollar Amount of Statement to Allow Printing.(only applicable if Allow Statement to be Excluded from Printing is enabled)')
     ),
 
+    'insurance_statement_exclude' => array(
+       xl('Do Not Print Statements For Insurance Companies'),
+       array(
+            '0' => xl('Primary'),
+            '1' => xl('Secondary'),
+            '2' => xl('Tertiary'),
```
Comment (aethelwulffe) [inline library/globals.inc.php]: Multi-select?  Not looking at the test (doing that tomorrow) but does the logic work out with the default that is only prints out the statement for Primary?

File: sites/default/statement.inc.php [hunk 2]
```diff
@@ -104,6 +104,12 @@ function create_HTML_statement($stmt) {
   #minimum_amount_due_to _print
   if ($stmt['amount'] <= ($GLOBALS['minimum_amount_to_print']) && $GLOBALS['use_statement_print_exclusion']) return "";
   if ($stmt['statement_print'] == "NO" && $GLOBALS['use_statement_print_exclusion']) return "";
+  if ($GLOBALS['use_statement_print_exclusion'] && $GLOBALS['insurance_statement_exclude'] !=4 ){
+   if ($GLOBALS['insurance_statement_exclude'] ==0 && $stmt['insurance_no_statement_print_pri']) return "";
```
Comment (aethelwulffe) [inline sites/default/statement.inc.php]: Could use better {} syntax, and nested statements.  Not sure if the 4 enum values cover the behavior needed.

File: library/globals.inc.php (hunk 1 above)
Comment (teryhill) [reply library/globals.inc.php]: No Multi select. If the primary is selected then no statement will be printed if that insurance is primary.

File: sites/default/statement.inc.php (hunk 2 above)
Comment (teryhill) [reply sites/default/statement.inc.php]: May need more for combinations.

Comment (aethelwulffe) [reply sites/default/statement.inc.php]: Terry, does this satisfy the real-world need, or do we need those other combos?
Can the user override these controls at the actual printing UI?

Comment (teryhill) [reply sites/default/statement.inc.php]: The actual UI for printing is pretty dumb . The request was to not print statements for a specific insurance company IE medicare. There is also another request being worked on to make the UI a little more intelligent and offer more printing abilities.

**Comment by aethelwulffe** [comment]
Terry, Tony, Others...
Are you guys OK with merging this now?  It has been sitting here for a while.  We need some reviews and such on this older material.

**Comment by aethelwulffe** [comment]
101 to 2.0 conflict keep popping up.  Editing in the middle of this file seems to cause these issues.

**Comment by teryhill** [comment]
Conflict resolved

**Comment by aethelwulffe** [comment]
I am testing again now.  I am learning that I need a new DB every time there is an upgrade....

**Comment by teryhill** [comment]
@aethelwulffe bump

**Comment by aethelwulffe** [comment]
This is in danger of going stale.  I think we are going to have to go forward with just one review.

**Pull request merged**